app/js/components/orderEntry/PastOrders.jsx

```diff
@@ -1,24 +1,55 @@
 import React from 'react';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
+import ReactPaginate from 'react-paginate';
 import { getPastOrders } from '../../actions/pastOrders';
 import PastOrder from './PastOrder';
 import imageLoder from '../../../img/loading.gif';
 
 export class PastOrders extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      pageCount: 0,
+      limit: 10,
+      startIndex: 0,
+      pageNumber: 0,
+    };
+    this.onPageChange = this.onPageChange.bind(this);
+  }
+
   componentDidMount() {
     this.getAllPastOrders(this.props);
   }
+
   componentWillReceiveProps(nextProps) {
     if (nextProps.tabName !== this.props.tabName) {
       this.getAllPastOrders(nextProps);
     }
+    this.setState({ pageCount: nextProps.pageCount });
+  }
+
+  onPageChange({ selected }) {
+    let { startIndex, pageNumber } = this.state;
+    const { limit } = this.state;
+
+    startIndex = Math.ceil(selected * limit);
+    pageNumber = startIndex / limit;
+
+    this.setState(() => ({ startIndex, pageNumber }));
+    this.getAllPastOrders(this.props, { limit, startIndex });
   }
-  getAllPastOrders=(props) => {
+
+  getAllPastOrders=(props, newState) => {
+    const { limit, startIndex = 0 } = newState || this.state;
+
+    this.setState(() => ({ startIndex: 0 }));
+
     const query = new URLSearchParams(this.props.location.search);
     const patientUuid = query.get('patient');
-    this.props.getPastOrders(patientUuid, props.careSetting.uuid);
+    this.props.getPastOrders(limit, startIndex, patientUuid, props.careSetting.uuid);
```

Comment:
Did you consider the fact that there will be a need to dispatch the getPastOrders from the add drug orders actions? I am wondering about all these parameters being passed.

Reply:
@Betty-Kebenei 
Was not aware of such implementation. 
However, based on the REST API design for OpenMRS, there are some global limitations applied to REST API resources. For instance, the default global result limit is 50. Because of this restriction, if you want to fetch more past orders you would have to paginate your results by setting the `limit` and `startIndex` values.